icu4j/main/classes/core/src/com/ibm/icu/text/DateTimePatternGenerator.java

```diff
@@ -2649,6 +2649,25 @@ DateTimeMatcher set(String pattern, FormatParser fp, boolean allowDuplicateField
                 if (subField > 0) subField += value.length();
                 type[field] = subField;
             }
+
+            // #20739, we have a skeleton with milliseconde, but no seconds
+            if (!original.isFieldEmpty(FRACTIONAL_SECOND) && original.isFieldEmpty(SECOND)) {
+                // Force the use of seconds
+                for (int i = 0; i < types.length; ++i) {
+                    int[] row = types[i];
+                    if (row[1] == SECOND) {
+                        // first entry for SECOND
+                        original.populate(SECOND, (char)row[0], (char)row[3]);
+                        baseOriginal.populate(SECOND, (char)row[0], (char)row[3]);
+                        // We add value.length, same as above, when type is first initialized.
+                        // The value we want to "fake" here is "s", and 1 means "s".length()
+                        int subField = row[2];
```

Comment:
According to the `types` table at the bottom of the file, `row[2]` is `NUMERIC`, which is defined as `0x100`, so I think you can just assign `type[SECOND] = row[2]` directly, like they're doing in the next loop with `DAYPERIOD`.  Also makes it more consistent with C++.

Reply:
Made C++ consistent with Java (and itself)
Without adding that length the "mS" Java the unit tests fails.